systest/backup/s3-backup/backup_test.go

```diff
@@ -0,0 +1,496 @@
+				code
+				message
+			}
+	}`
+
+	// For restore we have to always use newly added restore cluster
+	adminUrl := "https://" + testutil.R_SockAddrHttp + "/admin"
+	params := testutil.GraphQLParams{
+		Query: restoreRequest,
+		Variables: map[string]interface{}{
+			"loc": backupDst,
+			"ak":  MapIGet[0],
+			"sk":  MapIGet[1],
+		},
+	}
+	b, err := json.Marshal(params)
+	require.NoError(t, err)
+
+	client := testutil.GetHttpsClient(t)
+	resp, err := client.Post(adminUrl, "application/json", bytes.NewBuffer(b))
+	require.NoError(t, err)
+	defer resp.Body.Close()
+
+	var data interface{}
+	require.NoError(t, json.NewDecoder(resp.Body).Decode(&data))
+
+	//require.Equal(t, "Success", testutil.JsonGet(data, "data", "restore", "code").(string))
+	receivedcode := testutil.JsonGet(data, "data", "restore", "code").(string)
+
+	//Waiting for 2 seconds, allowing restore to finish, since restore dosen't give us taskID and data is small
+	//For larger data/payload, please increase the waiting time.
+
+	time.Sleep(2 * time.Second)
+
+	return receivedcode
+}
+
+// runFailingRestore is like runRestore but expects an error during restore.
+func runFailingRestore(t *testing.T, backupLocation, lastDir string, commitTs uint64) {
+	// Recreate the restore directory to make sure there's no previous data when
+	// calling restore.
+	require.NoError(t, os.RemoveAll(restoreDir))
+
+	result := worker.RunRestore("./data/restore", backupLocation, lastDir, x.SensitiveByteSlice(nil), options.Snappy, 0)
+	require.Error(t, result.Err)
+	require.Contains(t, result.Err.Error(), "expected a BackupNum value of 1")
+}
+
+func dirSetup(t *testing.T) {
+	// Clean up data from previous runs.
+	dirCleanup(t)
+
+	for _, dir := range testDirs {
+		if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+			t.Fatalf("Error while creaing directory: %s", err.Error())
+		}
+	}
+}
+
+func dirCleanup(t *testing.T) {
+	MapIGet := getEnvs()
+
+	if err := os.RemoveAll("./data"); err != nil {
+		t.Fatalf("Error removing direcotory: %s", err.Error())
+	}
+
+	//Cleanup the s3 too
+
+	sess, _ := session.NewSession(&aws.Config{
+		Region: aws.String(MapIGet[3])},
+	)
+
+	// Create S3 service client
+	svc := s3.New(sess)
+
+	// Setup BatchDeleteIterator to iterate through a list of objects.
+	iter := s3manager.NewDeleteListIterator(svc, &s3.ListObjectsInput{
+		Bucket: aws.String(MapIGet[2]),
+	})
+
+	// Traverse iterator deleting each object
+	if err := s3manager.NewBatchDeleteWithClient(svc).Delete(aws.BackgroundContext(), iter); err != nil {
+		exitErrorf("Can't delete object(s) from bucket because %v", err)
+	}
+}
+
+func exitErrorf(msg string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, msg+"\n", args...)
+	os.Exit(1)
+}
+
+func getEnvs() [4]string {
+
+	S3_ACCESS_KEY := os.Getenv("S3_ACCESS_KEY")
+	S3_SECRET_KEY := os.Getenv("S3_SECRET_KEY")
+	S3_BUCKET_NAME := os.Getenv("S3_BUCKET_NAME")
+	S3_REGION := os.Getenv("S3_REGION")
```

Comment:
Not sure if it is best practice to keep S3 secrets stored in environment variables (e.g. [ref](https://blog.diogomonica.com/2017/03/27/why-you-shouldnt-use-env-variables-for-secret-data/)).  This information can be easily leaked.  Also I don't see where this information is being added to the environment.  @darkn3rd has more expertise in this area.